MTA-6811, 6812, 6813, 6814, 6815, 6816 CQA April Developer Lightspeed work (#357)
migtools/mta-documentation: docs/topics/developer-lightspeed/ref_example-code-suggestion.adoc

= Generating code fix suggestions 

[role="_abstract"]

> Generate code fixes for a Java application you want to migrate to the target technology `quarkus`. To stream code changes that resolve issues in the code, use the Agent mode and the `my-model` as the large language model (LLM) that you deployed in {ocp-name} AI.

Reviewer: ```suggestion
Generate code fixes for a Java application you want to migrate to the `quarkus` target technology. To stream code changes that resolve issues in the code, use the Agent mode and the `my-model` as the large language model (LLM) that you deployed in {ocp-name} AI.
```

> Generate code fixes for a Java application you want to migrate to the `quarkus` target technology. To stream code changes that resolve issues in the code, use the Agent mode and the `my-model` as the large language model (LLM) that you deployed in {ocp-name} AI.

Reviewer: ```suggestion
Generate code fixes for a Java application you want to migrate to the `quarkus` target technology. To stream code changes that resolve issues in the code, use the Agent mode and `my-model` as the large language model (LLM) that you deployed in {ocp-name} AI.
```

> This example will walk you through generating code fixes for a Java application you want to migrate to the target technology `quarkus`. To generate resolutions for issues in the code, we use the Agentic AI mode and the `my-model` as the large language model (LLM) that you deployed in {ocp-name} AI.

Reviewer 2: This is self-referential and uses future tense which we should avoid in docs. How about something along these lines?

```suggestion
Generate code fixes for a Java application you want to migrate to the target technology `quarkus`. Using Agentic AI mode and a deployed large language model (LLM) helps you automatically create resolutions for issues in your code.
```

Fit it if it occurs elsewhere too.

You must change the `provider-setting` configuration if you plan to use a different LLM provider.
====

. On the `{ProductShortName}: Open Analysis View` page, click *Start* to start the {ProductShortName} Remote Procedure Call (RPC) server.

. Select the profile you configured and click *Run Analysis* to scan the Java application.
+
{ProductShortName} identifies the issues in the code.

> . Click the solutions icon (image:solutions-icon-new.png[Click solutions icon]) in an issue to request suggestions to resolve the issue.

Reviewer 2: Alt text for images must be in quotation marks. See https://docs.google.com/document/d/13NAUVAby1y1qfT77QFIZrMBhi872e7IEvAC9MUpGXbQ/edit?tab=t.0

```suggestion
. Click the solutions icon (image:solutions-icon-new.png["Click solutions icon"]) in an issue to request suggestions to resolve the issue.
```

Fit it if it occurs elsewhere too.

+
{mta-dl-plugin} streams the issue description, a preview of the code changes in files that resolve the issue.
+
You can review the code changes in the editor and accept or reject the changes. If you accept the changes, {mta-dl-plugin} creates a new file with the accepted code changes.
+
. Click *Continue* to allow {mta-dl-plugin} to run a follow-up analysis. 
+

> This round of analysis detects lint issues, compilation issues, or diagnostic issues that can occur when you accepted the suggested code change.

Reviewer: ```suggestion
This round of analysis detects lint issues, compilation issues, or diagnostic issues that can occur when you accept the suggested code change.
```

You must change the `provider-setting` configuration if you plan to use a different LLM provider.
====

. On the `{ProductShortName}: Open Analysis View` page, click *Start* to start the {ProductShortName} Remote Procedure Call (RPC) server.

. Select the profile you configured and click *Run Analysis* to scan the Java application.
+
{ProductShortName} identifies the issues in the code.

. Click the solutions icon (image:solutions-icon-new.png["Click solutions icon"]) in an issue to request suggestions to resolve the issue.
+

> {mta-dl-plugin} streams the issue description, a preview of the code changes in files that resolve the issue.

Reviewer: ```suggestion
{mta-dl-plugin} streams the issue description and a preview of the code changes in files that resolve the issue.
```
